### PR TITLE
feat(select): customize remove/clear/menuItemSelected icons

### PR DESCRIPTION
**Pull request by chengcyber**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for branch `feature`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Related Issue: https://github.com/ant-design/ant-design/issues/12954

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/select/index.tsx [hunk 1]
```diff
@@ -190,24 +196,30 @@ export default class Select extends React.Component<SelectProps, {}> {
...
+    const finalClearIcon = clearIcon && (
+      React.isValidElement<{ className?: string }>(clearIcon)
+        ? React.cloneElement(clearIcon) : clearIcon) || (
       <Icon type="close-circle" theme="filled" className={`${prefixCls}-clear-icon`} />
     );
 
-    const menuItemSelectedIcon = (
+    const finalMenuItemSelectedIcon = menuItemSelectedIcon && (
+      React.isValidElement<{ className?: string }>(menuItemSelectedIcon)
+        ? React.cloneElement(menuItemSelectedIcon) : menuItemSelectedIcon) || (
```
Comment (zombieJ) [inline components/select/index.tsx]: Also need pass ```className={`${prefixCls}-selected-icon`}```:

**Comment by zombieJ** [comment]
Pls help to update related docs.

File: components/select/index.tsx [hunk 2]
```diff
@@ -190,24 +196,30 @@ export default class Select extends React.Component<SelectProps, {}> {
...
 
     return (
       <RcSelect
         inputIcon={inputIcon}
-        removeIcon={removeIcon}
-        clearIcon={clearIcon}
-        menuItemSelectedIcon={menuItemSelectedIcon}
+        removeIcon={finalRemoveIcon}
+        clearIcon={finalClearIcon}
+        menuItemSelectedIcon={finalMenuItemSelectedIcon}
```
Comment (afc163) [inline components/select/index.tsx]: 不需要这样转一遍吧，原来的 rest 里面会覆盖的。

File: components/select/index.tsx (hunk 1 above)
Comment (kimochg) [reply components/select/index.tsx]: After reading the code, I found the specified className of `suffixIcon` is not passed when using custom `suffixIcon`, while clear/remove/suffix icons have not their style. So, they work fine right now. In the case of the potentional style issue, all of the classNames are passed to the custom icons right now.

**Comment by kimochg** [comment]
@afc163 @zombieJ  Thanks for your code review. I append two more commits solving the className issue and update the doc.

File: components/select/index.en-US.md [hunk 3]
```diff
@@ -47,6 +47,9 @@ Select component to select value from options.
 | showSearch | Whether show search input in single mode. | boolean | false |
 | size | Size of Select input. `default` `large` `small` | string | default |
 | suffixIcon | The custom suffix icon | ReactNode | - |
+| removeIcon | The custom remove icon | ReactNode | - |
+| clearIcon | The custom clear icon | ReactNode | - |
+| menuItemSelectedIcon | The custom menuItemSelected icon | ReactNode | - |
```
Comment (afc163) [inline components/select/index.en-US.md]: Do document there props cause we don't encourage people to customize them.

Comment (kimochg) [reply components/select/index.en-US.md]: hmm, so do I need to remove them?

Comment (afc163) [reply components/select/index.en-US.md]: yep

Comment (kimochg) [reply components/select/index.en-US.md]: @afc163  done. the commit of updating doc has been dropped now.

**Comment by kimochg** [comment]
master rebased.

**Comment by afc163** [comment]
image

I don't think you are doing right.

**Comment by kimochg** [comment]
```
// origin is this repo
git fetch origin master
git rebase origin/master
```
That's what I run.

Let's make it clear. Do you mean rebase feature branch instead of master?
Now, I revert back and rebase the feature branch to make the commits correct.

**Comment by afc163** [comment]
Sorry, you should rebase feature branch. It is my mistake of the comment above.

**Comment by kimochg** [comment]
I rebased the latest feature branch, but still ci broken. After checking the ci logs, some doesn't come from my code. But I do change the select dom structure, and new select demo snapshots have been updated now.

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/commit/a33e7a41d7d57c94ee7da3d23ec2be7c0d693275

Rebase again, fixed it right now.

**Comment by kimochg** [comment]
update:
ci broken. It seems to me that a new property `context` added into Checkbox, Sider, Menu, Radio components, it breaks both lint & compile ci.

**Comment by afc163** [comment]
Rebased again 😆

**Comment by kimochg** [comment]
rebased, a new bunch of errors...:scream:

**Comment by afc163** [comment]
Rebase again 😆...

**Comment by kimochg** [comment]
sth changed in `transfer` and `time-picker` break two checks :-\

**Pull request merged**